ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java

```diff
@@ -338,4 +373,42 @@ private void addStatTask(ASTNode root, Table table, Path oldPartitionLocation, P
       moveTask.addDependentTask(statTask);
     }
   }
+
+  public StringBuilder constructDeleteQuery(Table table, Map<String, String> partitionSpec) throws SemanticException {
+    StringBuilder sb = new StringBuilder().append("delete from ").append(table.getTableName())
+            .append(" where ");
+    List<String> keyList = new ArrayList<String>(partitionSpec.keySet());
+    Deserializer deserializer = table.getDeserializer();
+    Map<String, PrimitiveObjectInspector.PrimitiveCategory> stringTypeInfoMap = new HashMap<>();
+    try {
+      ObjectInspector objectInspector = deserializer.getObjectInspector();
+      if (objectInspector.getCategory() == ObjectInspector.Category.STRUCT) {
+        StructObjectInspector structObjectInspector = (StructObjectInspector) objectInspector;
+        List<? extends StructField> structFields =  structObjectInspector.getAllStructFieldRefs();
+        for (int index = 0;index < structFields.size();index++) {
+          StructField structField = structFields.get(index);
+          if (structField.getFieldObjectInspector().getCategory() == ObjectInspector.Category.PRIMITIVE) {
+            PrimitiveObjectInspector primitiveObjectInspector = (PrimitiveObjectInspector) structField.getFieldObjectInspector();
+            stringTypeInfoMap.put(structField.getFieldName(),
+                primitiveObjectInspector.getTypeInfo().getPrimitiveCategory());
+          }
+        }
+      }
+    } catch (SerDeException e) {
+      throw new SemanticException(String.format("Unable to get object inspector due to: %s", e));
+    }
+    for (int index = 0;index < keyList.size();index++) {
+      String key = keyList.get(index);
+      PrimitiveObjectInspector.PrimitiveCategory category = stringTypeInfoMap.get(key);
+      String value = partitionSpec.get(key);
+      boolean shouldEncloseQuotes = TypeInfoUtils.shouldEncloseQuotes(category);
+      sb.append(index == 0 ? "" : " and ").append(key).append(" = ");
+      if (shouldEncloseQuotes) {
+        sb.append("'").append(value).append("'");
```

Comment:
Does `'` has to be escaped?
```
table's_partition
```
```
pkey = 'table''s_partition'
```

Reply:
Changed to - 
`sb.append("\'").append(value).append("\'");`